AllReadyApp/Web-App/AllReady/Areas/Admin/Features/Campaigns/LockUnlockCampaignCommandHandler.cs

```diff
@@ -0,0 +1,37 @@
+﻿using AllReady.Models;
+using MediatR;
+using System.Linq;
+
+namespace AllReady.Areas.Admin.Features.Campaigns
+{
+    public class LockUnlockCampaignCommandHandler : RequestHandler<LockUnlockCampaignCommand>
+    {
+        private AllReadyContext _context;
+
+        public LockUnlockCampaignCommandHandler(AllReadyContext context)
+        {
+            _context = context;
+
+        }
+        protected override void HandleCore(LockUnlockCampaignCommand message)
+        {
+            var campaign = 
+                _context.Campaigns.SingleOrDefault(c => c.Id == message.CampaignId);
+
+            if (campaign != null)
+            {
+                if (campaign.Locked)
```

Comment:
@stevejgordon Sorry, I tend to leave comments on the first line of a block instead of the last forgetting how GitHub likes to show them. What I mean is that
``` CSharp
if (campaign.Locked)
{
  campaign.Locked = false;
}
else
{
  campaign.Locked = true;
}
```
is functionally equivalent to and more succinctly written as
```
campaign.Locked = !campaign.Locked;
```